pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -159,8 +159,10 @@ func (r *PerformanceProfileReconciler) hypershiftReconcile(ctx context.Context,
 		//REVIEW - Is it ok to use the ctx Context with the remote client???
 		if err := hypershiftDeleteComponents(r.Client, ctx, instance); err != nil {
 			klog.Errorf("[%s] Error: failed to delete components. err: %v", req.NamespacedName, err)
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "Deletion failed", "[hypershift:%s] Failed to delete components: %v", req.NamespacedName, err)
```

Comment:
events are not bad, I'm in favor of them. But still setting conditions (especially for degraded) seems worthy

Reply:
I am totally agree with that. 
The problem here is that PerformanceProfile is embedded inside a Configmap which is owned by the Hypershift's NodePoolController, so if I change anything of the configmap it will be reconciled again by the NodePoolController.